src/modules/pages/components/RouteLayouts/Complex/Complex.tsx

```diff
@@ -0,0 +1,62 @@
+import React, { ReactNode } from 'react';
+import { useLocation } from 'react-router-dom';
+
+import { RouteComponentProps } from '~pages/RouteLayouts';
+import SubscribedColoniesList from '~dashboard/SubscribedColoniesList';
+import SimpleNav from '../SimpleNav';
+import HistoryNavigation from '../HistoryNavigation';
+
+import styles from './Complex.css';
+
+interface Props {
+  children: ReactNode;
+  routeProps?: RouteComponentProps;
+}
+
+const displayName = 'pages.Complex';
+
+const Complex = ({
+  children,
+  routeProps: {
+    hasBackLink = true,
+    backRoute,
+    backText,
+    backTextValues,
+    hasSubscribedColonies = true,
+  } = {},
+}: Props) => {
+  const location = useLocation<{ hasBackLink?: boolean }>();
+  const backLinkExists =
+    hasBackLink === undefined
```

Comment:
!hasBackLink ?  
isn't enough?

Reply:
No, because you also set this, not just by props, but also by `react-router`'s state, so if it's `undefined` check the state, otherwise, use whatever the prop value is